### PR TITLE
Remove configsuite

### PR DESCRIPTION
**Pull request by oyvindeide**

### CODE REVIEW

File: stea/stea_config.py [hunk 1]
```diff
@@ -1,171 +1,115 @@
...
-def _fix_keys(elem):
-    fix_dict = {}
-    for key in elem:
-        new_key = str(key).replace("-", "_")
-        if key != new_key:
-            print(f"Warning: Replacing deprecated {key} with {new_key}")
-        fix_dict[new_key] = elem.get(key)
-    return fix_dict
 
+class EclProfile(BaseModel):
```
Comment (berland) [inline stea/stea_config.py:14]: Rename to e.g. SimulatorProfile?

File: stea/stea_config.py [hunk 2]
```diff
@@ -1,171 +1,115 @@
...
+        alias_generator=replace_dashes,
+    )
+    data: Optional[conlist(float, min_length=1)] = Field(
+        description="Values",
+    )
+
+
+class SteaConfig(BaseModel):
+    config_date: datetime = Field(
+        description="timestamp: Y-M-D H:M:S that comes with stea request",
```
Comment (berland) [inline stea/stea_config.py:65]: Spell out YYYY-MM-DD HH:MM:SS ?

File: stea/stea_config.py [hunk 3]
```diff
@@ -1,171 +1,115 @@
...
+        description=(
+            "The id (a number) of the project, which must already "
+            "exist and be available in the stea database. In the "
+            "Stea documentation this is called 'AlternativeId'"
+        ),
+        alias_generator=replace_dashes,
+    )
+    project_version: int = Field(
+        description=(
+            "Project alternative version number that " "comes from stea database"
```
Comment (berland) [inline stea/stea_config.py:78]: join strings

File: stea/stea_config.py [hunk 4]
```diff
@@ -1,171 +1,107 @@
...
+        description=(
+            "The profiles keyword is used to enter profile data explicitly in the "
+            "configuration file. Each profile is identified with an id from the "
+            "existing stea project, a start year and the actual data."
+        ),
+        alias_generator=replace_dashes,
+    )
+    ecl_profiles: Dict[str, SimulatorProfile] = Field(
+        description=(
+            "Profiles which are calculated directly from an eclipse simulation. "
```
Comment (sondreso) [inline stea/stea_config.py:86]:
```suggestion
            "Profiles which are calculated directly from a reservoir simulation. "
```

File: stea/stea_config.py [hunk 5]
```diff
@@ -1,171 +1,107 @@
...
+    start_date: Optional[date] = Field(
+        None,
+        description=(
+            "By default fmu-steaclient will calculate a profile from the full "
+            "time range of the simulated data, but you can optionally use the keywords "
+            "start_date and end_year to limit the time range. Production on the "
+            "start_date is included. Date format is YYYY-MM-DD."
+        ),
+    )
+    start_year: Optional[int] = Field(None, description="Deprecated. Use start_date")
```
Comment (sondreso) [inline stea/stea_config.py:25]: The visibility of this deprecation warning changes a bit with this. I guess this is what we really want: https://github.com/pydantic/pydantic/pull/8237

**Review by sondreso — APPROVED** [review]
Except for the small comments, LGTM :ship:

**Review by berland — APPROVED** [review]

**Pull request merged**